server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -702,6 +702,10 @@ private String validateAdd(DiscoveryNode node) {
             return null;
         }
 
+        public Map<String, DiscoveryNode> getNodes() {
```

Comment:
This seems to be added just for a test. I think that test can call build() to get the DiscoveryNodes instance and call getNodes() on that?

Reply:
sure, this requires a slight change to other tests using buildNodes but agree that it should look better